tests/test_examples.py

```diff
@@ -34,39 +36,69 @@ def get_file_paths():
     return test_files
 
 
-def create_checker(test_file, checker_name):
-    """Creates a test function from a test file, and a checker name.
-    test_file: The full path (string) to the file.
-    checker_name: The hyphenated checker name that should be detected.
-    An example of a valid checker_name would be: 'no-init-classes'
+def run_pylint_on_examples() -> Dict[str, str]:
+    """Return a dict mapping example file path to its pylint output
     """
-    # The following are captured when this function is created.
-    def new_test_func():
+
+    pylint_output = {}
+
+    file_paths = get_file_paths()
+    full_output = _get_full_pylint_output(file_paths)
+    _, *individual_outputs = full_output.split("*************")
```

Comment:
You can investigate how to use pylint's `json` reporter so that you get back JSON. This way you should be able to eliminate all string parsing, and simplify this function. You shouldn't need to define any helper functions.